etc/esp-idf/README.md

```diff
@@ -0,0 +1,7 @@
+# ESP-IDF Build Instructions
```

Comment:
please consider adding an `idf_component.yml` in this directory with the following contents:
```
dependencies:
  idf: "<5.3"
```
This will restrict compilation of OpenMRN to ESP-IDF version 5.2.x or earlier by default until I can get a PR setup for the driver updates in ESP-IDF v5.3

Reply:
We don't use the IDF component manager for several reasons:
- it becomes impossible to compile the code when the espressif servers are not reachable over the internet (have you tried compiling code while you are on an airplane?)
- the compilation becomes not deterministic from the git hash

Both of these are big problems for a professional project. Reproducible compilation results are essential for finding bugs and retroactive maintenance. We are talking about 10y+ of support needs in this product domain, not a 6-month product development cycle like many other domains.

Having said that, it should be possible to use ther component manager if someone wants it. It is just a requirement for the compilation and dependencies to work correctly if someone chooses not to use it.

Reply:
Reproducible builds is a great call-out and has a rather simple solution. Limit your external dependencies (beyond esp-idf itself) *AND* embed the github link and hash (which typically can not change) and the build system will only use that specific version for external dependencies.

As for offline (airplane) usage, this is doable with small adjustments to the idf_component.yml and pre-staging the external dependencies locally. In the idf_component.yml file you would leverage the `override_path` or `path` element to shortcut the lookup of the component with a local path to where the dependency actually lives. You might need to remove version/git/etc elements though (I forget now on that).

Cloning the dependencies locally into a components tree (as git submodules) would also work for this, optionally adding the following snippet to the top level CMakeLists.txt file:
```
list(APPEND EXTRA_COMPONENT_DIRS "../common_components/Filesystem/")
```
This allows simplified dependencies mapping in component / main CMakeLists.txt files.

Reply:
also note that the above snippet would *ONLY* check the ESP-IDF version and does not specifically deal with dependencies. If the ESP-IDF version check does not match the declared version it will abort the build with a message for the user to fix their ESP-IDF version selection.

Reply:
@atanisoft I've added an idf_component.yml file. This causes a warning for my non-managed build, but we can choose to ignore it. Can you have a look and provide some feedback?

Reply:
commented on the file above, we should just include the basic snippet listed above to only restrict the idf version check.

Reply:
@bakerstu it looks like no variant of declaration in idf_component.yml will work correctly with the current version of the idf component manager that I can find. The `path: etc/esp-idf` bit tries to force the git checkout to that path rather than reference that path for where to find the top level directory.

I suspect it only applies to when publishing the component to their registry and not for when consuming it. For now let's drop the file and I'll see if I can find a suitable way to make it work correctly or log an issue for them to fix it